clients/drcov/postprocess/drcov2lcov.cpp

```diff
@@ -675,7 +675,7 @@ search_cb(drsym_info_t *info, drsym_error_t status, void *data)
         strstr(info->name, op_test_pattern.get_value().c_str()) != NULL) {
         char *name = (char *)malloc(strlen(info->name) + 1);
         /* strdup is deprecated on Windows */
-        strncpy(name, info->name, strlen(info->name) + 1);
+        strcpy(name, info->name);
```

Comment:
So in this case g++ emits a warning saying:

`‘char* strncpy(char*, const char*, size_t)’ specified bound depends on the length of the source argument`

It also detects this if you create a variable with the length, use that to allocate name, and then use it as the size for strncpy. This could be avoided by using strcpy or memcpy (in this case they are equivalent), though I don't know if memcpy is considered acceptable. Do you have any suggestions or alternate implementations for this?

Reply:
Inside core/ we have dr_strdup.  For here: maybe dr_snprintf of %s?

Reply:
Done